swarm64_tpc_toolkit/correctness.py

```diff
@@ -27,14 +26,22 @@ def has_differences(self, first_df, second_df):
         if first_df.empty != second_df.empty:
             return True
 
-        diff = first_df.merge(second_df, indicator='source', how='outer')
-        self.diff = diff[diff['source'] != 'both']
-        self.diff['source'] = self.diff['source'].apply(
-            lambda x: 'benchmark results' if x == 'left_only' else 'correctness results')
-        diff_rows_count = self.diff.shape[0]
+        first_df = first_df.round(2)
```

Comment:
i'm really wondering why this is so much better than transforming the results beforehand (as it was before)

Reply:
IMO this here is more readable and makes use of existing functionality.